custom_components/fraimic/art_packs.py

```
@@ -319,6 +495,54 @@ async def async_install(self, pack_id: str) -> dict[str, Any]:
+    async def _async_materialize_pack(
+        self, pack: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Resolve a lazy Reframed row into a bounded current image list."""
+        provider_path = pack.get("provider_path")
+        if not isinstance(provider_path, str) or not provider_path:
+            return pack
+        if pack["images"]:
+            return pack
+        entries = loaded_fraimic_entries(self.hass)
+        if not entries:
+            raise HomeAssistantError("No Fraimic frame is loaded")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)